dipy/reconst/tests/test_dki.py

```diff
@@ -129,3 +140,268 @@ def test_dki_predict():
+
+    # Wyyzz
+    Wfit[1, 1, 2, 2] = Wfit[1, 2, 1, 2] = Wfit[1, 2, 2, 1] = kt_cross[11]
+    Wfit[2, 1, 1, 2] = Wfit[2, 2, 1, 1] = Wfit[2, 1, 2, 1] = kt_cross[11]
+
+    # Wxxyz
+    Wfit[0, 0, 1, 2] = Wfit[0, 0, 2, 1] = Wfit[0, 1, 0, 2] = kt_cross[12]
+    Wfit[0, 1, 2, 0] = Wfit[0, 2, 0, 1] = Wfit[0, 2, 1, 0] = kt_cross[12]
+    Wfit[1, 0, 0, 2] = Wfit[1, 0, 2, 0] = Wfit[1, 2, 0, 0] = kt_cross[12]
+    Wfit[2, 0, 0, 1] = Wfit[2, 0, 1, 0] = Wfit[2, 1, 0, 0] = kt_cross[12]
+
+    # Wxyyz
+    Wfit[0, 1, 1, 2] = Wfit[0, 1, 2, 1] = Wfit[0, 2, 1, 1] = kt_cross[13]
+    Wfit[1, 0, 1, 2] = Wfit[1, 1, 0, 2] = Wfit[1, 1, 2, 0] = kt_cross[13]
+    Wfit[1, 2, 0, 1] = Wfit[1, 2, 1, 0] = Wfit[2, 0, 1, 1] = kt_cross[13]
+    Wfit[2, 1, 0, 1] = Wfit[2, 1, 1, 0] = Wfit[1, 0, 2, 1] = kt_cross[13]
+
+    # Wxyzz
+    Wfit[0, 1, 2, 2] = Wfit[0, 2, 1, 2] = Wfit[0, 2, 2, 1] = kt_cross[14]
+    Wfit[1, 0, 2, 2] = Wfit[1, 2, 0, 2] = Wfit[1, 2, 2, 0] = kt_cross[14]
+    Wfit[2, 0, 1, 2] = Wfit[2, 0, 2, 1] = Wfit[2, 1, 0, 2] = kt_cross[14]
+    Wfit[2, 1, 2, 0] = Wfit[2, 2, 0, 1] = Wfit[2, 2, 1, 0] = kt_cross[14]
+
+    # Function to be tested
+    Wcons = dki.Wcons(kt_cross)
+
+    Wfit = Wfit.reshape(-1)
+    Wcons = Wcons.reshape(-1)
+
+    assert_array_almost_equal(Wcons, Wfit)
+
+
+def test_spherical_dki_statistics():
+    # tests MK solutions are equal to an expected values for a spherical
+    # kurtosis tensor
+
+    # Define multi voxel spherical kurtosis simulations
+    MParam = np.zeros((2, 2, 2, 27))
+    MParam[0, 0, 0] = MParam[0, 0, 1] = MParam[0, 1, 0] = params_sph
+    MParam[0, 1, 1] = MParam[1, 1, 0] = params_sph
+    # MParam[1, 1, 1], MParam[1, 0, 0], and MParam[1, 0, 1] remains zero
+
+    MRef = np.zeros((2, 2, 2))
+    MRef[0, 0, 0] = MRef[0, 0, 1] = MRef[0, 1, 0] = Kref_sphere
+    MRef[0, 1, 1] = MRef[1, 1, 0] = Kref_sphere
+    MRef[1, 1, 1] = MRef[1, 0, 0] = MRef[1, 0, 1] = 0
+
+    # Mean kurtosis analytical solution
+    MK_multi = mean_kurtosis(MParam)
+    assert_array_almost_equal(MK_multi, MRef)
+
+    # Mean kurtosis numerical method
+    sph = Sphere(xyz=gtab.bvecs[gtab.bvals > 0])
+    MK_multi = mean_kurtosis(MParam, sph)
+    assert_array_almost_equal(MK_multi, MRef)
+
+    # radial kurtosis analytical solution
+    RK_multi = radial_kurtosis(MParam)
+    assert_array_almost_equal(RK_multi, MRef)
+    
+    # axial kurtosis analytical solution
+    AK_multi = axial_kurtosis(MParam)
+    assert_array_almost_equal(AK_multi, MRef)
+
+
+def test_compare_MK_method():
+    """ tests if analytical solution of MK is equal to the exact solution"""
+
+    # OLS fitting
+    dkiM = dki.DiffusionKurtosisModel(gtab_2s)
+    dkiF = dkiM.fit(signal_cross)
+
+    # MK analytical solution
+    MK_as = dkiF.mk
+
+    # MK numerical method
+    sph = Sphere(xyz=gtab.bvecs[gtab.bvals > 0])
+    MK_nm = mean_kurtosis(dkiF.model_params, sph)
+
+    assert_array_almost_equal(MK_as, MK_nm, decimal=1)
```

Comment:
When a sphere is given as input,  MK is estimated as the average of the directional kurtosis of the sphere's vertices instead of the MK's analytical solution (see documentation dipy.reconst.dki.mean_kurtosis). This is why computations can be so different.